src/main/kotlin/io/github/mojira/arisa/modules/MultiplePlatformsModule.kt

```diff
@@ -0,0 +1,68 @@
+package io.github.mojira.arisa.modules
+
+import arrow.core.Either
+import arrow.core.extensions.fx
+import arrow.core.left
+import arrow.core.right
+import io.github.mojira.arisa.domain.Comment
+import io.github.mojira.arisa.domain.Issue
+import io.github.mojira.arisa.domain.Link
+import java.time.Instant
+
+class MultiplePlatformsModule(
+    private val platformWhitelist: List<String>,
+    private val targetPlatform: String,
+    private val transferredPlatformBlacklist: List<String>,
+    private val keepPlatformTag: String
+) : Module {
+    override fun invoke(issue: Issue, lastRun: Instant): Either<ModuleError, ModuleResponse> = with(issue) {
+        Either.fx {
+            assertPlatformWhitelisted(platform, platformWhitelist).bind()
+            assertTrue(isDuplicatedWithDifferentPlatforms(platform, transferredPlatformBlacklist, issue).bind()).bind()
+            assertNotKeepPlatformTag(comments).bind()
+            updatePlatforms(targetPlatform)
+        }
+    }
+
+    private fun isDuplicatedWithDifferentPlatforms(platform: String?, blacklist: List<String>, issue: Issue):
+            Either<ModuleError, Boolean> = Either.fx {
+        issue.links
+            .filter(::isDuplicatedLink)
+            .forEach {
+                val child = it.issue.getFullIssue().toFailedModuleEither().bind()
+                if (child.platform !in blacklist && child.platform != platform.getOrDefault("None")) {
+                    return@fx true

```

Comment:
I don't think you need a FX block here, its really heavy on the call stack(3 levels causes a stackoverflow)

Reply:
That's how it's written in the ConfirrmParentModule

Reply:
That's where I took it from

Reply:
maybe both shouldnt have it :P

Reply:
the only reason is there is                 val child = it.issue.getFullIssue().toFailedModuleEither().bind()

You can just do an if for that (without toFailedModuleEither)

Reply:
without doesn't exist? Plus, for some reason the platform and reporter fields are recognized by the code unless .bind() is attached to the end of the val child line

Reply:
I'm going to leave as-is right now unless its very necessary to do this

Reply:
when I said without I mean removing .toFailedModuleEither() and just doing the if yourself

By the way, it doesnt really make sense to convert it to a either anyways

Reply:
getFullIssue() already returns an either, just a different one

and I don't see a cleaner way to deal with that honestly

and bind is needed 'cos when Either is Right it converts Either into its right half, that Boolean that you need